src/pages/home/HeaderView.tsx

```diff
@@ -144,8 +144,8 @@ function HeaderView({report, personalDetails, parentReport, policy, session, rep
     );
 
     const canJoinOrLeave = !isSelfDM && !isGroupChat && (isChatThread || isUserCreatedPolicyRoom || canLeaveRoom);
-    const canJoin = canJoinOrLeave && !isWhisperAction && props.report.notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN;
-    const canLeave = canJoinOrLeave && ((isChatThread && props.report.notificationPreference.length) || isUserCreatedPolicyRoom || canLeaveRoom);
+    const canJoin = canJoinOrLeave && !isWhisperAction && report.notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN;
+    const canLeave = canJoinOrLeave && ((isChatThread && report.notificationPreference?.length) ?? isUserCreatedPolicyRoom ?? canLeaveRoom);
```

Comment:
```suggestion
    const canLeave = canJoinOrLeave && ((isChatThread && report.notificationPreference?.length) || isUserCreatedPolicyRoom || canLeaveRoom);
```

If notificationPreference length is 0 we want to check the rest of the conditions